bundler/lib/bundler/cli/gem.rb

```diff
@@ -155,15 +155,16 @@ def run
         templates.merge!("CHANGELOG.md.tt" => "CHANGELOG.md")
       end
 
-      if ask_and_set(:rubocop, "Do you want to add rubocop as a dependency for gems you generate?",
-        "RuboCop is a static code analyzer that has out-of-the-box rules for many " \
-        "of the guidelines in the community style guide. " \
-        "For more information, see the RuboCop docs (https://docs.rubocop.org/en/stable/) " \
-        "and the Ruby Style Guides (https://github.com/rubocop-hq/ruby-style-guide).")
-        config[:rubocop] = true
-        config[:rubocop_version] = Gem.ruby_version < Gem::Version.new("2.4.a") ? "0.81.0" : "1.7"
+      config[:linter] = ask_and_set_linter
+      case config[:linter]
+      when "rubocop"
+        config[:linter_version] = Gem.ruby_version < Gem::Version.new("2.4.a") ? "0.81.0" : "1.7"
         Bundler.ui.info "RuboCop enabled in config"
         templates.merge!("rubocop.yml.tt" => ".rubocop.yml")
+      when "standard"
+        config[:linter_version] = "0.2.5" if Gem.ruby_version < Gem::Version.new("2.4.a")
```

Comment:
(I  made this comment in my initial review but somehow I deleted it, reposting it now)

Would it be too bad if we locked the version in general, at least until you reach 1.0? Users might get an outdated standard version by default, but we'll probably get updates proposed and we'll try to keep it up to date.

Our CI is hard enough to keep green, so we don't want it to also break to due to changes in third party libraries.

Reply:
I take your point. Here are some of my thoughts:

As a consultant, I visit lots and lots of Ruby teams, and something that's very common (but thanks to Bundler's sophistication, usually unnecessary) is for teams to proactively lock down versions of everything in their Gemfile and only make updates harder for themselves. When a version is pinned in a Gemfile, it (should) communicate to the reader "there is something dangerous about upgrading this and trusting the lockfile", but if most or every gem is locked by default (as it is in the Gemfile generated by Rails, too), then a pinned version doesn't tell the reader anything useful; it just makes staying up-to-date harder. I think tools that generate Gemfiles (chief of them Bundler, of course) should lead by example and default to only pinning versions where there's a known risk of upgrading, to help people get and stay in the habit of staying better on-top-of their dependencies by trusting bundle update and version control to role back any future incompatibilities

As for the risk to Bundler's CI (because I can empathize having worked to get all 72 checks green), the integrations I've added with standard that could break things are (1) a name change, (2) a change to the require statement in the Rakefile, (3) the name of the standard rake command, and (4) they default style ruleset as it applies to the gemspec (which is a pretty minimal file). At least for items 1-3, I'm so confident that these won't change that I'm willing to commit to proactively fix them in Bundler before they do.

To your third point, we haven't come to decide what 1.0 will "mean" from Standard's perspective (e.g. should it be a commitment that code fixes shouldn't churn git without a major bump? Obviously Rubocop isn't adhering to that, so that'd be tough). But one thing I will say is that the road from here to wherever 1.0 is to increase its stability and to relax its rules that cause false negatives, so in practice I wouldn't recommend anybody pin their Gemfile to a version now (and pinning it in this tool is essentially a context-free recommendation, re: the above). If this ends up causing you all any grief, I'm willing to come back and offer some kind of solution to stabilize your build (including pinning the version in tests somehow without leaking that through to the end user)

Are you comfortable moving forward on this as-is, or do you want it pinned before merge?

Reply:
I agree with your points.

The key is each one interpretation of "should lead by example and default to only pinning versions where there's a known risk of upgrading". I think it's a common (and reasonable) conception that gems in their 0.x series are inherently unstable, so pinning those in the default template seems totally reasonable to me. And given we don't pin other gems in there, I think it's easy to understand why the gem is being pinned.

Regarding the risk to bundler's CI, (4) is exactly the issue I had in mind, reinforced by the fact that standard warns about this on its own README.

I'd say let's pin it.

Reply:
>  (4) they default style ruleset as it applies to the gemspec (which is a pretty minimal file)

Just to clarify, it's not only the gemspec but all files in the default gem skeleton, right? Definitely not a lot, but more than a single small file, and they might grow.

Reply:
> As a consultant, I visit lots and lots of Ruby teams, and something that's very common (but thanks to Bundler's sophistication, usually unnecessary) is for teams to proactively lock down versions of everything in their Gemfile and only make updates harder for themselves.

I can confirm I've seen this too, and I'm willing to improve our documentation with better Gemfile best practices that most people can agree with. I bet if you properly explain this to teams currently doing this, they would change their practice.